picard/ui/infodialog.py

```diff
@@ -88,8 +92,56 @@ def _display_artwork_tab(self):
             if image.width and image.height:
                 infos.append(u"%d x %d" % (image.width, image.height))
             infos.append(image.mimetype)
-            item.setText(u"\n".join(infos))
-            self.ui.artwork_list.addItem(item)
+
+            img_wgt = self.artwork_table.get_coverart_widget(pixmap, "\n".join(infos))
+            self.artwork_table.setCellWidget(row, col, img_wgt)
+            self.artwork_table.setItem(row, col, item)
+            row += 1
+
+    def _display_artwork_type(self):
+        types = []
+        types = [image.types_as_string() for image in self.obj.metadata.images]
+        if self.display_existing_artwork:
+            existing_types = [image.types_as_string() for image in self.obj.orig_metadata.images]
+            #Merge both lists
```

Comment:
I first printed all the types in *Type* column (using `_display_artwork_type`), and then iterated through cover images (using `_display_artwork`) and draw image in a cell if image type matches the type in corresponding row in *Type* column. 
To print the types, I needed to take the union of two lists (types in new covers and types in existing covers). Sets won't work as they remove duplicates and some albums may have duplicate cover types (like multiple front). So, I wrote a merge instance. If a type is same in both lists, append it only once (`else` case), and rest is obvious.
After types are printed, first new cover images are iterated. Only those image are drawn whose type matches with corresponding type in *Type* column. Then existing cover images are iterated and drawn.